src/main/java/com/google/devtools/build/lib/bazel/bzlmod/Discovery.java

```diff
@@ -91,6 +91,12 @@ public static Result run(Environment env, RootModuleFileValue root)
           moduleFileValue =
               (ModuleFileValue) result.getOrThrow(skyKey, ExternalDepsException.class);
         } catch (ExternalDepsException e) {
+          if (e.getDetailedExitCode().getFailureDetail() == null
+              || e.getDetailedExitCode().getFailureDetail().getExternalDeps().getCode()
+                  != FailureDetails.ExternalDeps.Code.BAD_MODULE) {
+            // This is not due to a bad module, so don't print a dependency chain.
```

Comment:
I see. Could you please briefly explain this in the code comment too?

In general checks like these tend to become stale (e.g. when we add new error codes), but since this is only for error message generation, it's probably fine.